File: apps/server/src/modules/analytics/analytics.controller.ts
import type { Request, Response } from 'express';
import { analyticsService } from './analytics.service.js';
import { asyncHandler } from '../../utils/asyncHandler.js';
import { ok } from '../../utils/ApiResponse.js';

export const analyticsController = {
  ordersOverTime: asyncHandler(async (req: Request, res: Response) => {
    const days = Math.min(365, Math.max(1, Number(req.query.days) || 30));
    const data = await analyticsService.ordersOverTime(days);
    return ok(res, data, 'Orders over time');
  }),

  revenueByDeliveryType: asyncHandler(async (_req: Request, res: Response) => {
    const data = await analyticsService.revenueByDeliveryType();
    return ok(res, data, 'Revenue by delivery type');
  }),

  topDrivers: asyncHandler(async (_req: Request, res: Response) => {
    const data = await analyticsService.topDrivers();
    return ok(res, data, 'Top drivers');
  }),

  operationsMetrics: asyncHandler(async (req: Request, res: Response) => {
    const days = Math.min(365, Math.max(1, Number(req.query.days) || 30));
    const data = await analyticsService.operationsMetrics(days);
    return ok(res, data, 'Operations metrics');
  }),

  satisfactionSummary: asyncHandler(async (req: Request, res: Response) => {
    const days = Math.min(365, Math.max(1, Number(req.query.days) || 90));
    const data = await analyticsService.satisfactionSummary(days);
    return ok(res, data, 'Satisfaction summary');
  }),
};

File: apps/guzo-api/src/main/java/et/guzo/web/controller/AnalyticsController.java
package et.guzo.web.controller;

import et.guzo.common.ApiResponse;
import et.guzo.service.AnalyticsService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/analytics")
@RequiredArgsConstructor
public class AnalyticsController {

    private final AnalyticsService analyticsService;

    @GetMapping("/orders-over-time")
    public ApiResponse<List<Map<String, Object>>> ordersOverTime(@RequestParam(defaultValue = "30") int days) {
        return ApiResponse.ok(analyticsService.ordersOverTime(days));
    }

    @GetMapping("/revenue-by-type")
    public ApiResponse<List<Map<String, Object>>> revenueByType() {
        return ApiResponse.ok(analyticsService.revenueByType());
    }

    @GetMapping("/top-drivers")
    public ApiResponse<List<Map<String, Object>>> topDrivers(@RequestParam(defaultValue = "5") int limit) {
        return ApiResponse.ok(analyticsService.topDrivers(limit));
    }

    @GetMapping("/operations-metrics")
    public ApiResponse<Map<String, Object>> operationsMetrics(@RequestParam(defaultValue = "30") int days) {
        return ApiResponse.ok(analyticsService.operationsMetrics(days));
    }

    @GetMapping("/satisfaction")
    public ApiResponse<Map<String, Object>> satisfaction(@RequestParam(defaultValue = "90") int days) {
        return ApiResponse.ok(analyticsService.satisfactionSummary(days));
    }
}
